packages/react/accordion/src/Accordion.tsx

```diff
@@ -37,46 +67,136 @@ type AccordionOwnProps = Polymorphic.Merge<
      * `defaultValue` if you do not need to control the state of an accordion.
      */
     defaultValue?: string;
-    /**
-     * Whether or not an accordion is disabled from user interaction.
-     *
-     * @defaultValue false
-     */
-    disabled?: boolean;
     /**
      * The callback that fires when the state of the accordion changes.
      */
     onValueChange?(value: string): void;
   }
 >;
 
-type AccordionPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof Primitive>,
-  AccordionOwnProps
+type AccordionSinglePrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof AccordionImpl>,
+  AccordionSingleOwnProps
 >;
 
-/**
- * `Accordion` is the root component.
- */
-const Accordion = React.forwardRef((props, forwardedRef) => {
+const AccordionSingle = React.forwardRef((props, forwardedRef) => {
   const {
     value: valueProp,
     defaultValue,
-    disabled,
     onValueChange = () => {},
-    ...accordionProps
+    ...accordionSingleProps
   } = props;
 
-  const buttonNodesRef = React.useRef<Set<React.ElementRef<typeof AccordionButton>>>(new Set());
-  const accordionRef = React.useRef<React.ElementRef<typeof Accordion>>(null);
-  const composedRefs = useComposedRefs(accordionRef, forwardedRef);
-
   const [value, setValue] = useControlledState({
     prop: valueProp,
     defaultProp: defaultValue,
-    onChange: (value) => value && onValueChange(value),
+    onChange: onValueChange,
+  });
+
+  return (
+    <AccordionValueProvider
+      value={value ? [value] : []}
```

Comment:
Nice, so from the context perspective it's always an array?

Reply:
You got it :+1: